### DISCUSSION TITLE
FileUploads with Laravel Sail and Minio

### DISCUSSION THREAD

**Discussion by AlexandreGerault** [filamentphp/filament#6826] [Help] [answered]
Hi!

I've a Crud resource where the form needs a picture. My development environment is a Laravel Sail environment and we're using a Minio container for the S3 server on local (keeps things clearer than a shared bucket on AWS). Thing is the `FileUpload` needs to upload directly from the frontend to the s3 server.

As we're using Minio in a container, the url to access it from the php container is `http://{minio-container-name}:9000` (which is `http://minio:9000` in our case). Thus my env file look like this
```env
AWS_ACCESS_KEY_ID=minio
AWS_SECRET_ACCESS_KEY=minio
AWS_DEFAULT_REGION=eu-west-1
AWS_BUCKET=admin
AWS_ENDPOINT=http://minio:9000
AWS_URL=http://localhost:9000/admin
AWS_USE_PATH_STYLE_ENDPOINT=true
```
Also I created the `admin` bucket.

<img width="736" alt="image" src="https://github.com/filamentphp/filament/assets/3494871/d3fbbc10-1ce2-4140-92e3-fb6a688aedc1">
 
You can see above what I get when I try to upload a file with this setup. Changing the endpoint to `http://localhost:9000` (as it is the endpoint visible from the browser/host point of view) it works. Should I make two disks? 🤔 Is there an option to change it only for filament components uploading directly on the server? 🤔

**Comment by AlexandreGerault** [comment, marked as answer]
An easy fix would be to switch temporary file on a local disk if you're not in a production environment, like so:
```php
return [
    'temporary_file_upload' => [
        'disk' => env('APP_ENV') === 'production' ? 's3' : 'local',
    ],
];
```

**Reply by spinanicky** [reply]
@AlexandreGerault where would this code go? 
Edit: Found the answer by reviewing here: https://github.com/filamentphp/filament/issues/9215#issuecomment-1800738948

Effectively create a livewire.php file in your config folder.

**Comment by brainplusplus** [comment]
i have tutorial on medium :
https://muhammadtriwibowo.medium.com/configuring-minio-filamentphp-and-filament-breezy-in-lara-b260c54be9a0
